### PR TITLE
Introducing card(), value_box(), and layout_column_wrap()

### PR DESCRIPTION
**Pull request by cpsievert**
Adds a new `card()` API as well as `value_box()` and `layout_column_wrap()`. To learn more about this new functionality, refer to these new pkgdown articles

* https://rstudio.github.io/bslib/articles/cards.html 
* https://rstudio.github.io/bslib/articles/value-boxes.html 
* https://rstudio.github.io/bslib/articles/layouts.html 

To achieve [responsive sizing](https://rstudio.github.io/bslib/articles/cards.html#responsive-sizing) (i.e., `card_body_fill()`), this PR also adds `.vfill-container`/`.vfill-item` classes that htmlwidgets (and shiny) will include to allow `shiny::plotOutput()`, `htmlwidgets::shinyWidgetOutput()`, etc., to fill by default when inside of `card_body_fill()`. Here are those relevant PRs:

https://github.com/rstudio/shiny/pull/3708
https://github.com/ramnathv/htmlwidgets/pull/437

Also, since many output containers that wrap `htmlwidgets::shinyWidgetOutput()` (e.g., `plotly::plotlyOutput()`) have  "hard-coded" `height` and `width` values in their function signatures, we'll be changing that as well (in as many places as we can -- it'd be great to have help from the community on this):

https://github.com/plotly/plotly.R/pull/2192
https://github.com/rstudio/leaflet/pull/822
https://github.com/rstudio/DT/pull/1026

### Testing notes

Automated tests are being added in https://github.com/rstudio/shinycoreci/pull/132

Since those automated tests are primarily screenshots, it'd also be great to manually test these apps on Firefox, Safari, and Edge, as well as deployed on hosted environments. Please reach out to me to get an overview of what behavior to expect.

https://github.com/rstudio/bslib/tree/main/inst/examples/card
https://github.com/rstudio/bslib/tree/main/inst/examples/value_box

### CODE REVIEW

File: inst/components/value-box.scss [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  .value-box-showcase {
+    flex: 1 1 auto;
+
+    // TODO: is this the best way to override the inline
+    // height/width on output bindings?
+    > * {
+      width: var(--showcase-full-width, 100%) !important;
+      height: var(--showcase-full-height, 600px) !important;
+    }
+  }
```
Comment (cpsievert) [inline inst/components/value-box.scss:72]: This should probably be redone with the vfill stuff (And should users have to opt-in the to the fill behavior)?
